server/api/crud/runtimes/nuclio/function.py

```diff
@@ -378,6 +381,17 @@ def _compile_function_config(
     )
 
     _resolve_and_set_base_image(function, config, client_version, client_python_version)
+    build: mlrun.model.ImageBuilder = function.spec.build
+    base_image = mlrun.utils.get_in(config, "spec.build.baseImage")
+
+    if (
+        base_image
+        and server.api.utils.builder.is_mlrun_image(base_image)
+        and build.requirements
+    ):
+        server.api.utils.builder.add_mlrun_to_requirements(build, base_image)
```

Comment:
This section can also be moved inside `_resolve_and_set_build_requirements_and_commands` as it is also setting build requirements (+ it keeps the code even cleaner)